SampleScripts/ppm/passwordpolicymanager.ps1

```diff
@@ -0,0 +1,1441 @@
+				Write-LogMessage -Type INFO -Message "Attempting to connect to NSX-T server '$server'"
+				$NSXToken = Request-NsxToken -fqdn $server -username $user -password $pass 
+				if ($NSXToken) {
+					Write-LogMessage -Type INFO -Message "Connected to NSX-T server '$server'"
+					Write-LogMessage -Type INFO -Message "Retrieving password policy from NSX-T server '$server'"
+					$response = Get-NsxtManagerAuthPolicy -nsxtManagerNode $server
+					$nsxtConfigTypeValues.passwdMinimumLength = $response.minimum_password_length
+					$nsxtConfigTypeValues.apiPasswdMaxFailAttempts = $response.api_max_auth_failures
+					$nsxtConfigTypeValues.apiPasswdMaxFailIntervalInSec = $response.api_failed_auth_reset_period
+					$nsxtConfigTypeValues.apiPasswdUnlockIntervalInSec = $response.api_failed_auth_lockout_period
+					$nsxtConfigTypeValues.cliPasswdMaxFailAttempts = $response.cli_max_auth_failures
+					$nsxtConfigTypeValues.cliPasswdMaxFailIntervalInSec = $response.cli_failed_auth_lockout_period    
+
+					$nodeConfigValue = New-Object -TypeName psobject
+					$nodeConfigValue | Add-Member -notepropertyname "FQDN" -notepropertyvalue $server
+					$nodeConfigValue | Add-Member -notepropertyname "productType" -notepropertyvalue "NSXMgr"
+					$nodeConfigValue | Add-Member -notepropertyname "productAttributes" -notepropertyvalue $nsxtConfigTypeValues
+					$passwordPolicyNodes.Add($nodeConfigValue)
+					return $passwordPolicyNodes
+				} else {
+					if ($ErrorLog -match '"error_code":404') {
+						$count += 1
+						Write-LogMessage -Type WARNING -Message "Unable to submit request. Retry again in 20 secs (Try $count/3)"
+						Start-Sleep -s 20
+					} else {
+						Write-LogMessage -Type INFO -Error "Failed to obtain access token from NSX-T server '$server',  please verify your credentials and try again"  -ErrorAction Stop
+					}
+				}
+			}
+			Write-LogMessage -Type INFO -Error "Testing a connection to server $server failed, please check your details and try again" -ErrorAction Stop
+		} else {
+			Write-LogMessage -Type INFO -Error "Testing a connection to server $server failed, please check your details and try again" -ErrorAction Stop
+		}
+    }
+    Catch {
+        Debug-ExceptionWriter -object $_
+    }
+	Finally {
+		Write-LogMessage -Type INFO -Message "Retrieving NSX-T server '$server' password policies completed"
+	}
+}
+
+Function GetNsxtEdgeNode {
+	<#
+        Retrieving NSX Edge Node details from Transport Node ID
+    #>
+	Param (
+        [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String]$transportNodeID
+    )
+    Try {
+        $uri = "https://$nsxtmanager/api/v1/transport-nodes/$transportNodeID"
+        $response = Invoke-RestMethod -Method GET -URI $uri -ContentType application/json -headers $nsxtHeaders
+        $response
+    }
+    Catch {
+        Write-Error $_.Exception.Message
+    }
+}
+
+Function GetNsxtEdgeNodePasswordPolicy {
+	<#
+        Retrieve NSX Edge Node servers' Password Policy
```

Comment:
"NSX Node Password..."

Reply:
addressed in 99029d5ec8015e0c5c645e9abc7c3384e1b3091e